src/coreclr/jit/flowgraph.cpp

```diff
@@ -5972,17 +5980,73 @@ bool FlowGraphNaturalLoop::CanDuplicate(INDEBUG(const char** reason))
     }
 #endif
 
-    Compiler*       comp   = m_dfsTree->GetCompiler();
-    BasicBlockVisit result = VisitLoopBlocks([=](BasicBlock* block) {
-        if (!BasicBlock::sameEHRegion(block, GetHeader()))
+    Compiler*         comp   = m_dfsTree->GetCompiler();
+    BasicBlock* const header = GetHeader();
+
+    CompAllocator           alloc = comp->getAllocator(CMK_LoopClone);
```

Comment:
```suggestion
    CompAllocator           alloc = comp->getAllocator(CMK_TryRegionClone);
```
would this make more sense?

Reply:
Yes, fixed.